elfo-core/src/messages.rs

```diff
@@ -81,12 +83,41 @@ pub struct ConfigUpdated {
 #[non_exhaustive]
 pub struct Terminate {
     pub(crate) closing: bool,
+    pub reason: Option<TerminateReason>,
```

Comment:
Why do we need `Option` if `TerminateReason` has the `Unknown` variant?

Reply:
We don't, initially i thought that `Option<TerminationReason>` without `Unkown` would be a way to go, but i like flatten struct more.